pkg/api/service/scans.go

```diff
@@ -339,64 +354,49 @@ func (s ScansService) ProcessScanCheckNotification(ctx context.Context, msg []by
 		return nil
 	}
 
-	// Don't take into account inconsistent progress in a message with a terminal status.
-	if checkStates.IsTerminal(checkMssg.Status) && (checkProgress != 1.0) {
-		_ = level.Error(s.logger).Log("FixingInvalidProgressInTerminalStatus", checkProgress, "Status", checkMssg.Status)
-		checkProgress = 1
-		checkMssg.Progress = &checkProgress
+	_, err = s.db.UpsertCheck(scanID, checkID, checkMssg, api.CheckStates.LessOrEqual(checkMssg.Status))
+	if err != nil {
+		return err
+	}
+	// As a check message does not contain all the information
+	// of a check we must merge with the the info of the check in the DB.
+	check := mergeChecks(dbCheck, checkMssg)
+	if err != nil {
+		return err
+	}
+	programID := ""
+	if check.Tag != nil {
+		programID = *check.Tag
 	}
+	s.notifyCheck(check, programID)
```

Comment:
The problem that I see with current implementation is that we are pushing metrics for each check message every time.
That should happen only once per status.

Reply:
Yep that's true, but if the message is have the same status for the checkid it does not matter right. The think is that we can not really avoid doing that because if for whatever reason a message is reprocessed we can not know if we notified the check before or not!! At least as far as I see it!!

Reply:
But I think that situation, like pushing exactly one with a "best effort" manner, assuming that we can lose some metrics in "edge cases" situations, is better than directly pushing incorrect metrics no? Otherwise the metrics will be useless at least for the RUNNING status.
What should we do here?